pkg/operator/controller/controller.go

```diff
@@ -184,30 +184,32 @@ func (r *reconciler) enforceEffectiveIngressDomain(ci *ingressv1alpha1.ClusterIn
 
 // haTypeForInfra returns the appropriate HA type for the given infrastructure
 // config.
-func haTypeForInfra(infraConfig *configv1.Infrastructure) ingressv1alpha1.ClusterIngressHAType {
+func haTypeForInfra(infraConfig *configv1.Infrastructure) operatorv1.EndpointPublishingStrategyType {
 	switch infraConfig.Status.Platform {
 	case configv1.AWSPlatform:
-		return ingressv1alpha1.CloudClusterIngressHA
+		return operatorv1.LoadBalancerServiceStrategyType
 	}
-	return ingressv1alpha1.UserDefinedClusterIngressHA
+	return operatorv1.PrivateStrategyType
 }
 
 // enforceEffectiveHighAvailability determines the effective HA configuration
 // for the given clusteringress and infrastructure config and publishes it to
 // the clusteringress's status.
-func (r *reconciler) enforceEffectiveHighAvailability(ci *ingressv1alpha1.ClusterIngress, infraConfig *configv1.Infrastructure) error {
+func (r *reconciler) enforceEffectiveHighAvailability(ci *operatorv1.IngressController, infraConfig *configv1.Infrastructure) error {
 	// The clusteringress's HA configuration is immutable, so if we have
 	// published an HA type status, we must continue using it.
-	if len(ci.Status.HighAvailability.Type) > 0 {
+	if ci.Status.EndpointPublishingStrategy != nil {
 		return nil
 	}
 
 	updated := ci.DeepCopy()
 	switch {
-	case ci.Spec.HighAvailability != nil:
-		updated.Status.HighAvailability.Type = ci.Spec.HighAvailability.Type
+	case ci.Spec.EndpointPublishingStrategy != nil:
+		updated.Status.EndpointPublishingStrategy = ci.Spec.EndpointPublishingStrategy
```

Comment:
Would be safer to use `.DeepCopy()` here.

Reply:
Good catch, fixed